File: files/200-299/206/CH-206.R
library(tidyverse)
library(readxl)

path = "files/CH-206 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:E7")

result = input %>%
  mutate(rn = row_number()) %>%
  separate_rows(ID, sep = "") %>%
  filter(ID != "") %>%
  mutate(pos = ifelse(row_number() %% 2 == 0, "Even Positions", "Odd Positions"), .by = rn) %>%
  pivot_wider(names_from = pos, values_from = ID, values_fn = ~ paste0(.x, collapse = "")) %>%
  select(-rn)

all.equal(result, test, check.attributes = FALSE)
# There is one mistake in result provided.

File: files/200-299/206/CH-206.py
import pandas as pd

path = "CH-206 Column Splitting.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=6)

def split_even_odd_chars(series):
    return series.apply(lambda x: pd.Series([''.join(str(x)[::2]), ''.join(str(x)[1::2])], index=['Odd Positions', 'Even Positions']))

result = split_even_odd_chars(input.iloc[:, 0])


print(test)
print(result)